test/test.erl

```diff
@@ -33,15 +33,17 @@ gen_should_fail() ->
         fun() ->
             %% user_types.erl is referenced by opaque_fail.erl.
             %% It is not in the sourcemap of the DB so let's import it manually
-            gradualizer_db:import_erl_files(["test/should_pass/user_types.erl"])
+            gradualizer_db:import_erl_files(["test/should_pass/user_types.erl"]),
+            %% exhaustive_user_type.erl is referenced by exhaustive_remote_user_type.erl
+            gradualizer_db:import_erl_files(["test/should_fail/exhaustive_user_type.erl"])
```

Comment:
You could use `user_types:my_opaque()` in the new test cases. Then you don't need to import another module here.

Reply:
Good idea, I'll try reworking the tests this way :+1:

Reply:
Thinking about it again, I think it makes sense to leave it the way it is.

Tests in `exhaustive_remote_user_type` mirror those from `exhaustive_user_type`. If an `exhaustive_user_type` test generates warnings, its counterpart in `exhaustive_remote_user_type` should generate the same or a similar warning. This wouldn't be easy to replicate if `exhaustive_remote_user_type` referred to tests in `user_types`, since the latter is a `should_pass` type test, whereas `exhaustive_user_type` is a `should_fail` type test. From yet another perspective, it makes sense to make `exhaustive_remote_user_type` and `exhaustive_user_type` work on the same types, but access them respectively as remote or local - it's just easier to reason about.